MetaMask/metamask-extension
# Added navigation between multiple sign prompts and reject all sign prompts

## Explanation

With this fix, the users should be able to navigate between multiple sign prompts back and forth and should be able to reject all sign prompts at once for `Eth Sign`, `Personal Sign`, `SignTypedData`, `SignTypedData_v3` and `SignTypedData_v4`. Also we are using a single component for navigation which will show both messages and transactions requests.

* Fixes #16848 
* Blocked by #16822 (ConfirmPageContainerNavigation refactor as it mentioned [here](https://github.com/MetaMask/metamask-extension/pull/16822#discussion_r1047452658))

## Screenshots/Screencaps

### Before

https://user-images.githubusercontent.com/92527393/211017881-9149191c-9a8f-47bf-bb7a-33fa7b1c6bb8.mp4

### After

#### popup view

https://user-images.githubusercontent.com/92527393/211017166-8b848b77-a2c4-4c61-a154-ecd08cc7e93e.mov

#### expanded view

https://user-images.githubusercontent.com/92527393/211015295-6cc2c555-5265-49a3-857b-a8f0b64cd19a.mov

### Using a single component for navigation which will show both messages and transactions requests and visa versa

https://user-images.githubusercontent.com/92527393/214558962-d9645d5c-9e76-4055-997d-074107f4c67e.mov

## Manual Testing Steps

1. Connect to Test Dapp https://metamask.github.io/test-dapp/
2. Click multiple times on `Eth Sign` or `Personal Sign` or `SignTypedData` or `SignTypedData_v3`or `SignTypedData_v4`.
3. The users should be able to navigate between multiple sign prompts back and forth and should be able to reject all sign prompts at once

### Using a single component for navigation which will show both messages and transactions requests and visa versa

1. Connect to Test Dapp https://metamask.github.io/test-dapp/
2. Click multiple times on `Piggy bank contract` section (`Deploy Contract`) or `Send Tokens` section  (`Create Token`) or `Send Eth` section (`Send Legacy Transaction` or `Send EIP 1559 Transaction`) and than click multiple times on `Eth Sign` or `Personal Sign` or `SignTypedData` or `SignTypedData_v3`or `SignTypedData_v4`.
3. The users should be able to navigate between multiple sign prompts and transactions back and forth
